### PR TITLE
fix: unable to un-indent a list item

### LINKED ISSUE #2466: Unable to un-indent a list item
### Description

Isn't possible to indent to the left in regular view (different behaviour in other view modes)

- [x] Can you reproduce the issue?

### Steps to reproduce

1. Create list
2. Indent first item then indent again for second item
3. Un-indenting a third item (moving it back to item #1 indentation) doesn't work with either the `shift + tab` or `ctrl + [` shortcuts. It seems in order to properly indent to the left by any amount, you have to move down one line (using down arrow key) and manually re-indent.

**Expected behavior:**

Before:
- Item1
    - Item 2
    - Item 3

After:
- Item1
    - Item 2
- Item 3

**Actual behavior:**

Before:
- Item 1
    - Item 2
    - Item 3

After:
- Item 1
    - Item 2
    -     Item 3 (same bullet level as item 2, just tabbed over one)

*Sidenote #1* - `ctrl + [` and `ctrl + ]` does absolutely nothing (in default view mode)

*Sidenote #2* - When in Source Code mode however, both `shift + tab` and `ctrl + [` function as expected

### Versions

- Mark Text version: 0.16.3.3214 (launched from .appimage executable instead of installed using .deb installer)
- Operating system: Linux Mint (cinnamon desktop)

Edit: formatting

### PR DESCRIPTION
**Pull request by orosmatthew**
fixes issue #2466 where shift-tab did not un-indent a list item

| Q                 | A
| ----------------- | ---
| Bug fix?          | yes
| New feature?      | no
| Breaking changes? | no
| Deprecations?     | no
| New tests added?  | not needed
| Fixed tickets     | Fixes #2466 
| License           | MIT

### Description

Fixes issue of not being able to shift-tab to un-indent list item

### CODE REVIEW

**Review by ermath — APPROVED** [review]

**Review by ermath — APPROVED** [review]
Looks good.

**Comment by ermath** [comment]
@orosmatthew, I wonder what it takes for Github to accept my approval and mark your PR as reviewed.

**Comment by orosmatthew** [comment]
I'm not 100% sure as this is my first pull request but it does state that "At least 1 approving review is required by reviewers with <strong>write access</strong>." I'm not sure how to check that but I guess you could check if you have this type of 'write access' to the repo.

**Review by ermath — APPROVED** [review]
Still looking good. :)

**Comment by ermath** [comment]
> Still looking good. :)

Hmm, I think all my permissions are in place. I've even forked the project at committed changes to a test branch. I don't know what else it takes to have my approval accepted by the system. I would really like to see this fix merged. :)

**Comment by orosmatthew** [comment]
Could it be the branch that I am trying to create a pull request with? I assumed I should use the marktext:develop branch but maybe permissions are set on a per-branch basis. Although I am not super familiar with how Github or this project specifically works.

**Comment by ermath** [comment]
> Could it be the branch that I am trying to create a pull request with? I assumed I should use the marktext:develop branch but maybe permissions are set on a per-branch basis. Although I am not super familiar with how Github or this project specifically works.

It could be that there needs to be a pull request from your feature branch into _your_ develop branch. And then some fork-automation-behind-the-scenes-magic updates the marktext:develop branch.

**Comment by Jocs** [comment]
Thanks for your contribution, it looks good.

**Review by Jocs — APPROVED** [review]

**Pull request merged**
